packages/redux-dynostore-react-redux/README.md

```diff
@@ -44,6 +44,30 @@ export default dynamic('identifier', somethingDynamic(), { context: CustomReactR
 
 This option overrides the `context` used for accessing the store in the React context.  If you are overriding this value, you should also be overriding the context passed to any `Provider` and/or `connect` components as well.  Please refer to the [Redux documentation](https://react-redux.js.org/using-react-redux/accessing-store#providing-custom-context) for more details on this use case.
 
+### `DynamicProvider`
+
+`dynamic` components will not render when server-side rendering (SSR) your app unless you add a `DynamicProvider` somewhere near the root of of your component tree (anywhere above the the first `dynamic` component).
```

Comment:
Maybe we should go further and recommend to make sure it is never mounted twice? (otherwise that useEffect will be reset and that `firstRender` will be set to `true` as every mount).

Reply:
I wonder if we set a flag in the global scope rather than in component state if that issue would go away?  Something like:

```js
let firstRender = true

export const DynamicProvider = ({ children }) => {
  useEffect(() => {
    firstRender = false
  }, [])

  return <DynamicContext.Provider value={firstRender}>{children}</DynamicContext.Provider>
}
```

Reply:
Trying this, I'm seeing the warning again... I'll keep playing.

Reply:
The issue was that without a `setState` call, the provider is never updated so the children always get `true`.  This appears to work:

```js
let firstRender = true

export const DynamicProvider = ({ children }) => {
  const [_, forceUpdate] = useState(0)
  useEffect(() => {
    firstRender = false
    forceUpdate((renderCount) => renderCount + 1)
  }, [])

  return <DynamicContext.Provider value={firstRender}>{children}</DynamicContext.Provider>
}
```

It's a bit hacky though

Reply:
I would recommend against that because that will definitely break SSR. The server will generally generate multiple pages so you should avoid relying on global variables. Here, for instance, for the second page being generated, the value will never be true.

Reply:
gotcha, that makes sense.  Component state it is.  I might change it though to inherit the state from a parent provider in case anyone nests them.  As long as there is a stable provider at the top of the tree is should be fine.

```js
const DynamicContext = React.createContext(null)

export const DynamicProvider = ({ children }) => {
  const inheritedFirstRender = useContext(DynamicContext)
  const [firstRender, setFirstRender] = useState(inheritedFirstRender === null)
  useEffect(() => {
    setFirstRender(false)
  }, [])

  return <DynamicContext.Provider value={firstRender}>{children}</DynamicContext.Provider>
}
```